boskos/boskos_test.go

```diff
@@ -0,0 +1,660 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/test-infra/boskos/common"
+	"k8s.io/test-infra/boskos/ranch"
+)
+
+func MakeFakeClient(resources []common.Resource) *ranch.Ranch {
+	newRanch := &ranch.Ranch{
+		Resources: resources,
+	}
+
+	return newRanch
+}
+
+func TestAcquire(t *testing.T) {
```

Comment:
Consider adding a concurrency test (aka have 10 people request items and ensure only one finds one, or something like that)